FhirToDataLake/docs/Process FHIR extensions.md

```diff
@@ -0,0 +1,101 @@
+"birthPlaceExtension.valueAddress.country": "{{ birth_place_extension.valueAddress.country }}",
+"birthPlaceExtension.valueAddress.postalCode": "{{ birth_place_extension.valueAddress.postalCode }}",
+}
+{% endvalidate -%}
+
+```
+
+_JSON schema file:_
+
+```javascript
+{
+    "title": "Patient customized schema",
+    "type": "object",
+    "properties": {
+        "resourceType": { type": "string" },
+        "id": { "type": "string" },
+	"birthPlaceExtension.valueAddress.city": { type": "string" },
+	"birthPlaceExtension.valueAddress.state": { "type": "string" },
+	"birthPlaceExtension.valueAddress.country": { "type": "string" },
+	"birthPlaceExtension.valueAddress.postalCode": { "type": "string" }
+    },
+    "required": [ "id" ]
+}
+```
+
+**Note**:
+1.	The JSON schema files must be saved at **Schema** directory in the image.
+	 
+	 ![image](./assets/LiquidDirectory.png)
+	 
+2.	The “validate” tag in template is optional and we recommend using it in your liquid template.
+3.	To avoid potential issues, you can test your templates and schema files before deploying the analytics pipeline.
+
+### 2.	Push the customized schema to Azure Container Registry
+Refer [here](https://github.com/microsoft/FHIR-Converter/blob/main/docs/TemplateManagementCLI.md) to push the prepared schema to Azure Container Registry, later we will use the schema image reference from the Container Registry to deploy the analytics pipeline.
+
+### 3.	Deploy the Analytics pipeline with customized schema enabled
+Deploy the pipeline with [ARM template](https://github.com/microsoft/FHIR-Analytics-Pipelines/blob/main/FhirToDataLake/deploy/templates/FhirSynapsePipelineTemplate.json), set the parameter “Customized Schema” as “true”, and “Customized Schema Image Reference” as the reference of image where your schema is stored.
+
+ ![image](./assets/DeploymentTemplate.png)
+
+After deploying the pipeline, the Azure Function agent will try to pull the customized schema from the given image reference.
+
+### 4.	Provide access of the Container Registry to the Azure Function
+Go to the Azure Container Registry instance, assign **“AcrPull”** role to **principal account** of your created Azure Function.
+ 
+ ![image](./assets/AccessRole.png)
+
+ ![image](./assets/AssignAccess.png)
+
+
+### 5.	Verify customized data on the Storage
+
+Customized data will be generated to “{resource type}_Customized” folder on the Storage.
+
+_Example customized patient Parquet data on the Storage_
+
+ ![image](./assets/ExampleParquetData.png)
+
+ ![image](./assets/ExampleParquetData2.png)
+
+## Query customized data in Synapse Serverless SQL pool
+
+### 1.	Run the Powershell script to create the tables and views with customized schema image
+
+Browse to the scripts folder under this path (..\FhirToDataLake\scripts).
+
+Run the following PowerShell script.
+./Set-SynapseEnvironment.ps1 -SynapseWorkspaceName "Synapse name" -Storage   "Storage name" -CustomizedSchemaImage "Schema image reference"
```

Comment:
Maybe make the example as a code content by adding ``` around it.

```
./Set-SynapseEnvironment.ps1 -SynapseWorkspaceName "Synapse name" -Storage "Storage name" -CustomizedSchemaImage "Schema image reference"
```